Add E2E tests for the integration in the classic product editor

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

- Add `menu_order` properties to the variations data in `tests/e2e/config/default.json` to make them can be updated correctly via the classic product editor.
- Add E2E tests

#### 📌 Checklist (@eason9487)

- [x] Revert 253f16cdb14802ffe47314e62f57eb4381c4ffab before merging this PR

### Detailed test instructions:

1. `npm run dev` to build files
2. `npm run test:e2e` to see if the E2E test in headless mode can pass.
3. (Optional) `npm run test:e2e -- --ui` to inspect the E2E test in UI mode.
4. View the E2E test result on GitHub Actions: https://github.com/woocommerce/google-listings-and-ads/actions/runs/8645102656/job/23701605350?pr=2363

### Changelog entry

eason9487 committed:
Temporarily enable pull requests to trigger E2E Tests on GitHub Actions.

eason9487 committed:
Set the variable product tour of classic product editor to not display.

eason9487 committed:
Classic product editor E2E test: Prompt to Get Started when not yet finished onboarding

eason9487 committed:
Classic product editor E2E test: Hide plugin tab and meta box for unsupported product types

eason9487 committed:
Classic product editor E2E test: Check existence and availability of fields for simple product

eason9487 committed:
Classic product editor E2E test: Check existence of fields for variable and variation products

eason9487 committed:
Classic product editor E2E test: Channel visibility is disabled when hiding in product catalog

eason9487 committed:
Classic product editor E2E test: Change channel visibility and check its notice, status, and issues

eason9487 committed:
Classic product editor E2E test: Custom input - Select with text input

eason9487 committed:
Classic product editor E2E test: Custom input - Date and time inputs

eason9487 committed:
Classic product editor E2E test: Custom input - Non-negative integer input

eason9487 committed:
Classic product editor E2E test: Save all product attributes to simple product

eason9487 committed:
Classic product editor E2E test: Save all product attributes to variation product

eason9487 commented on .github/workflows/e2e-tests.yml:
```diff
@@ -5,8 +5,6 @@ on:
     branches:
       - "release/**"
   pull_request:
-    branches:
-      - "release/**"
```
Memo: This temporary change will be reverted before merging PR.

martynmjones reviewed (approved):
Hey @eason9487, thanks for expanding the E2E tests to cover the classic product editor.

Tested the branch locally and confirmed tests are passing and expectations all look correct so LGTM ✅

puntope commented on tests/e2e/utils/product-editor.js:
```diff
@@ -10,6 +10,243 @@ import * as api from './api';
+			return page.locator( '.gla_attributes_multipack_field input' );
+		},
+
+		getAllProductAttributes( locator = page ) {
```
I see this function is almost the same as the `getProductBlockEditorUtils` version. Can we refactor it?

eason9487 replied:
Adjusted in https://github.com/woocommerce/google-listings-and-ads/pull/2363/commits/8a7993308650aae932702368c37b2e9160da2922.

puntope commented on tests/e2e/utils/product-editor.js:
```diff
@@ -10,6 +10,243 @@ import * as api from './api';
+		},
+
+		async gotoEditVariableProductPage() {
+			const variableId = await api.createVariableProduct();
+			await api.createVariationProducts( variableId );
```
This can be refactored as well.

eason9487 replied:
Adjusted in https://github.com/woocommerce/google-listings-and-ads/pull/2363/commits/8a7993308650aae932702368c37b2e9160da2922.

puntope commented on tests/e2e/specs/product-editor/classic-integration.test.js:
```diff
@@ -0,0 +1,707 @@
+			adultContent,
+		} = editorUtils.getAllProductAttributes( locator );
+
+		const allPairs = [
```
This is duplicated with block-integration equivalent. Can we refactor it?

eason9487 replied:
Adjusted in https://github.com/woocommerce/google-listings-and-ads/pull/2363/commits/f21dd78f0c3f994dcb1537a23b84c74fd69c4c09.

puntope reviewed (approved):
Thanks @eason9487 for your work here.

The tests are passing and the code looking good for me. 

I left some comments regarding possible refactors and preventing code duplication.

Approved in advance

eason9487 committed:
Create a new E2E API utils `createVariableWithVariationProducts` to reuse.

Address: https://github.com/woocommerce/google-listings-and-ads/pull/2363#discussion_r1563678082

eason9487 committed:
Extract the `getAllProductAttributes` in the E2E utils `product-editor` for reuse.

Address: https://github.com/woocommerce/google-listings-and-ads/pull/2363#discussion_r1563674580

eason9487 committed:
Extract the `getAvailableProductAttributesWithTestValues` to the E2E utils `product-editor` for reuse.

Address: https://github.com/woocommerce/google-listings-and-ads/pull/2363#discussion_r1563689276

eason9487 committed:
Revert "Temporarily enable pull requests to trigger E2E Tests on GitHub Actions."

This reverts commit 253f16cdb14802ffe47314e62f57eb4381c4ffab.
